include/nbl/ext/CentralLimitBoxBlur/CBlurPerformer.h

```diff
@@ -0,0 +1,131 @@
+    }
+
+    static inline uint32_t buildParameters(uint32_t numChannels, const asset::VkExtent3D& inputDimensions, Parameters_t* outParams, DispatchInfo_t* outInfos,
+        const float radius, const asset::ISampler::E_TEXTURE_CLAMP* wrappingType, const asset::ISampler::E_TEXTURE_BORDER_COLOR* borderColor = nullptr)
+    {
+        uint32_t passesRequired = 0u;
+        
+        if (numChannels)
+        {
+            for (uint32_t i = 0u; i < 3u; i++)
+            {
+                auto dim = (&inputDimensions.width)[i];
+                if (dim >= 2u)
+                    passesRequired++;
+            }
+        }
+
+        if (passesRequired)
+        {
+            for (uint32_t i = 0u; i < passesRequired; i++)
+            {
+                auto& params = outParams[i];
+
+                params.input_dimensions.x = inputDimensions.width;
+                params.input_dimensions.y = inputDimensions.height;
+                params.input_dimensions.z = inputDimensions.depth;
+        
+                assert(wrappingType[i] <= asset::ISampler::ETC_MIRROR);
+
+                const auto passAxis = i;
+                params.input_dimensions.w = (passAxis << 30) | (numChannels << 28) | (wrappingType[i] << 26);
+
+                if (borderColor)
+                {
+                    assert(borderColor[i] <= asset::ISampler::E_TEXTURE_BORDER_COLOR::ETBC_INT_OPAQUE_WHITE);
+                    params.input_dimensions.w |= borderColor[i] << 23;
+                }
+        
+                params.input_strides.x = 1u;
+                params.input_strides.y = inputDimensions.width;
+                params.input_strides.z = params.input_strides.y * inputDimensions.height;
+                params.input_strides.w = params.input_strides.z * inputDimensions.depth;
+
+                // At this point I'm wondering do you even need both input and output strides
+                params.output_strides = params.input_strides;
+
+                params.radius = radius;
+
+                auto& dispatch = outInfos[i];
+                dispatch.wg_count[0] = inputDimensions.width;
+                dispatch.wg_count[1] = inputDimensions.height;
+                dispatch.wg_count[2] = inputDimensions.depth;
+                dispatch.wg_count[passAxis] = 1u;
+            }
+        }
+        
+        return passesRequired;
+    }
+
+    static inline void dispatchHelper(video::IVideoDriver* driver, const video::IGPUPipelineLayout* pipelineLayout, const Parameters_t& params,
+        const DispatchInfo_t& dispatchInfo, bool issueDefaultBarrier = true)
+    {
+        driver->pushConstants(pipelineLayout, video::IGPUSpecializedShader::ESS_COMPUTE, 0u, sizeof(Parameters_t), &params);
+        driver->dispatch(dispatchInfo.wg_count[0], dispatchInfo.wg_count[1], dispatchInfo.wg_count[2]);
+
+        if (issueDefaultBarrier)
+            defaultBarrier();
+    }
```

Comment:
structure it like your blit filter's API

Reply:
Is adding memory barriers the user's responsibility?

Reply:
Just found this comment https://github.com/Devsh-Graphics-Programming/Nabla/pull/115#discussion_r926880360

Reply:
lets continue the discussion in that comment